### PR TITLE
 Add API guide for parameter.

### LINKED ISSUE #305: API Guide 第2期：模型平均、Gradient Clipping, Regulizer, 参数属性与参数初始化（11月19日初稿pr，11月26日定稿merge）
# API Guide写前须知

## 背景
API Guide可以理解为是API的使用指南，旨在向用户说明API的设计思路、使用方法等
撰写计划共计约[45篇](http://agroup.baidu.com/paddlepaddle/view/office/1136638)。其中在PaddlePaddle v1.1中，已上线14篇，例如[官网-API-API使用指南-卷积](http://paddlepaddle.org/documentation/api/zh/1.1/api_guides/low_level/layers/conv.html)

## 任务说明
一篇完整的API Guide，需要包含以下几个内容：

1）对本篇主题的简介，例如：

``
卷积有两组输入：特征图和卷积核，依据输入特征和卷积核的形状、Layout不同、计算方式的不同，在Fluid里，有针对变长序列特征的一维卷积，有针对定长图像特征的二维(2D Conv)、三维卷积(3D Conv)，同时也有卷积计算的逆向过程，下面先介绍Fluid里的2D/3D卷积，再来介绍序列卷积。
``

2）对主题下每个API的介绍，例如

``
这个layer以一个mini batch的序列为输入，在每个序列内做softmax操作。其输出为一个mini batch相同shape的序列，但在序列内是经softmax归一化过的。 这个layer往往用于在每个sequence内做softmax归一化。
API Reference 请参考 sequence_softmax
``

- 完整例子：
[1.1已上线的API Guide](https://github.com/PaddlePaddle/FluidDoc/tree/release/1.1/doc/fluid/api/api_guides/low_level)

- 文档题目：请见标题

- 语言：汉语

- 格式：rst格式（因官网对rst格式支持较好，推荐用rst格式，如不了解rst，请在[这里](https://sphinx-doc-zh.readthedocs.io/en/latest/rest.html)学习）

- 字数：不限

- 效果预览：

为提高Review效率，请先在本地预览，确认格式正确再提交pr，请将预览效果截图贴在pr中，**不预览的pr不能合入**
预览工具的使用方法请见[这里](https://github.com/PaddlePaddle/FluidDoc#preview-with-ppo)

请注意：FluidDoc Repo的文档是通过Sphinx工具显示在官网上的。在Sphinx中，通过名为index.rst（或index_cn.rst等类似名称）的文件中的toctree来决定每一级显示哪几篇文档，在预览时，为了让文档能够正确预览，需要修改index，例如，如果新增一篇layers的API Guide，则需要修改`doc/fluid/api/api_guides/low_level/layers/index.rst` 这里的文件，并将自己文档的名字加入到index，如下图：

image

但在提交pr前，请不要git add 对index的修改，以避免多人同时修改这个文件引发conflict

- 提交分支：develop

- 提交位置：提交位置在`FluidDoc/doc/fluid/api/api_guides/low_level`

注：
layers文档，请提交到`FluidDoc/doc/fluid/api/api_guides/low_level/layers`
多机文档，请提交到`FluidDoc/doc/fluid/api/api_guides/low_level/cluster`

- pr提交截止日期：**请见标题**

- merge截止时间：**请见标题**

- 需要**特别注意**的格式问题

在一篇API Guide中，需要给出链接跳转至API Reference，[例如](http://paddlepaddle.org/documentation/api/zh/1.1/api_guides/low_level/layers/sequence.html#sequence-softmax)：

image

请注意：这里的超链接是用**内链**写成的，这样做的好处是，当某个API Reference的位置发生变化时，无需专门在文档里改超链接的路径，依然可以正确跳转。

用内链引用API Reference的方法： 以  ` :ref:<api_name>` 的格式引用reference（而不是其他格式：例如超链接、`xxx`等），`api_name`需与对应rst文件中的名字一致。

参考示例：https://raw.githubusercontent.com/PaddlePaddle/FluidDoc/develop_doc/doc/fluid/api_guides/low_level/optimizer/optimizer_all.rst

## 提交pr时的注意事项

- 贴预览效果

您可以选择以下两种方式展示预览效果：贴图或者给出预览网址，以方便我们检查格式

- 检查提交了几个文件

在预览工具时，需要修改index以确保文档可以显示在预览工具中，但在提交pr时，请去掉对index的修改，只提交您新写的文档，这是为了避免多人同时修改index文件引发conflict

- PR Reviewer：

为了提高Review效率，请在PR提交时增加Reviewer：一个PR的Reviewer由Peer Reviewer+ RD Reviewer+PM Reviewer组成，分别负责检查技术、格式的问题

Peer Reviewer ：1名，请找同方向的同学，例如NLP相关API请加kuke
RD Reviewer：luotao1
PM Reviewer：tink2123

- PR Labels：

为了方便管理API Guide进度，请在PR提交时增加Label `API Guide`

### PR DESCRIPTION
**Pull request by wanghaoshuang**
fix #305 

image
image
image

### CODE REVIEW

File: doc/fluid/api/api_guides/low_level/parameter.rst [hunk 1]
```diff
@@ -0,0 +1,167 @@
+..  _api_guide_parameter:
+
+#########
+Parameter
+#########
+
+我将模型中的weight和bias统称为parameter, 在fluid中对应class fluid.Parameter. fluid.Parameter继承自fluid.Variable, 是一种可持久化的variable. 模型的训练就是不断学习更新parameters的过程。Parameter相关的属性可以通过 :ref:`api_fluid_param_attr_ParamAttr` 来配置，可配置内容有：
```
Comment (luotao1) [inline doc/fluid/api/api_guides/low_level/parameter.rst]: - `我将模型中的weight和bias统称为parameter`，太口语化了。
- `class fluid.Parameter. fluid.Parameter`有两个` fluid.Parameter`
- `fluid.Variable, 是一种可持久化的variable. `，逗号/句号是英文的，要改成中文标点。

File: doc/fluid/api/api_guides/low_level/parameter.rst [hunk 2]
```diff
@@ -0,0 +1,167 @@
+..  _api_guide_parameter:
+
+#########
+Parameter
```
Comment (luotao1) [inline doc/fluid/api/api_guides/low_level/parameter.rst]: 标题请改成中文

File: doc/fluid/api/api_guides/low_level/parameter.rst [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+fluid通过 :code:`ParamAttr` 的 :code:`do_model_average` 属性设置单个parameter是否进行平均优化。
+示例如下：
+
+  .. code-block:: python
+
+      param_attrs = fluid.ParamAttr(name="fc_weight",
+                                do_model_average=true)
+      y_predict = fluid.layers.fc(input=x, size=10, param_attr=param_attrs)
+
+在miniBatch训练过程中，每个batch过后，都会更新一次parameters, paddle model average做的就是平均最近k次更新产生的parameters.
```
Comment (luotao1) [inline doc/fluid/api/api_guides/low_level/parameter.rst]: - parameters,和parameters. 英文逗号/句号改成中文
- paddle model average改成模型平均？

File: doc/fluid/api/api_guides/low_level/parameter.rst (hunk 1 above)
Comment (wanghaoshuang) [reply doc/fluid/api/api_guides/low_level/parameter.rst]: Thx. Done.

File: doc/fluid/api/api_guides/low_level/parameter.rst (hunk 2 above)
Comment (wanghaoshuang) [reply doc/fluid/api/api_guides/low_level/parameter.rst]: Thx. Done.

File: doc/fluid/api/api_guides/low_level/parameter.rst (hunk 3 above)
Comment (wanghaoshuang) [reply doc/fluid/api/api_guides/low_level/parameter.rst]: Thx. Done.

**Comment by luotao1** [comment]
LGTM

File: doc/fluid/api/api_guides/index.rst [hunk 4]
```diff
@@ -12,5 +12,4 @@ API使用指南
     low_level/metrics.rst
     low_level/model_save_reader.rst
     low_level/inference.rst
-  
-  
+    low_level/parameter.rst
```
Comment (shanyi15) [inline doc/fluid/api/api_guides/index.rst]: 为了避免多人同时修改这个index文件造成冲突，辛苦去掉对这个文件的修改，由我们最后统一加index

**Review by shanyi15 — COMMENTED** [review]
很棒～

File: doc/fluid/api/api_guides/index.rst (hunk 4 above)
Comment (wanghaoshuang) [reply doc/fluid/api/api_guides/index.rst]: Thx. Done.

**Review by shanyi15 — APPROVED** [review]
LGTM
